### ISSUE TITLE
yank analyze error for implicit solvent simulations

### ISSUE THREAD

**Issue by MehtapIsik** [choderalab/yank#701]
I am getting the following error when I try to analyze my implicit solvent simulations with the following command:
`yank analyze --store=output/experiments`

```
Traceback (most recent call last):
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/bin/yank", line 11, in <module>
    load_entry_point('yank==0.16.2', 'console_scripts', 'yank')()
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/cli.py", line 71, in main
    dispatched = getattr(commands, command).dispatch(command_args)
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/commands/analyze.py", line 76, in dispatch
    analyze.analyze_directory(args['--store'])
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/analyze.py", line 1152, in analyze_directory
    data[phase_name] = phase.analyze_phase()
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/analyze.py", line 845, in analyze_phase
    DeltaH_ij, dDeltaH_ij = self.get_enthalpy()
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/analyze.py", line 789, in get_enthalpy
    self._compute_enthalpy_and_entropy()
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/yank-0.16.2-py3.5-linux-x86_64.egg/yank/analyze.py", line 769, in _compute_enthalpy_and_entropy
    (f_k, df_k, H_k, dH_k, S_k, dS_k) = self.mbar.computeEntropyAndEnthalpy()
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/pymbar/mbar.py", line 1176, in computeEntropyAndEnthalpy
    dDelta_f_ij = self._ErrorOfDifferences(covf,warning_cutoff=warning_cutoff)
  File "/cbio/jclab/home/misik/opt/miniconda3/envs/yank_0.16.2_f598e2b/lib/python3.5/site-packages/pymbar/mbar.py", line 1406, in _ErrorOfDifferences
    d2[(np.any(d2) < warning_cutoff)] = 0.0
IndexError: in the future, 0-d boolean arrays will be interpreted as a valid boolean index
```
I don't get this error with explicit solvent simulations. My outfile is stored at Hal cluster: `/cbio/jclab/home/misik/hsa_yank/implicit/hsa_dansylamide_i02/analysis`

**Comment by jchodera** [comment]
I can't tell if this is indicating there are likely so few samples that the uncertainty matrix is singular or if this is a weird `pymbar` error we should address.

**Comment by jchodera** [comment]
@Lnaden : It looks like `yank analyze` is not printing out the number of uncorrelated samples per thermodynamic leg for @MehtapIsik's data. Was this feature removed?

**Comment by jchodera** [comment]
Or is this just the log output and other information is being printed to `stdout`?
```
2017-06-20 21:36:06,704: Reading energies...
2017-06-20 21:36:06,850: Done.
2017-06-20 21:36:06,850: Deconvoluting replicas...
2017-06-20 21:36:08,547: Done.
2017-06-20 21:36:09,045: Computing free energy differences...
2017-06-20 21:36:11,278: Cumulative symmetrized state mixing transition matrix:
2017-06-20 21:36:11,278:            0     1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17    18    19
2017-06-20 21:36:11,278:      0 0.555 0.245 0.103 0.051                                                                                                
2017-06-20 21:36:11,278:      1 0.245 0.290 0.202 0.122 0.067                                                                                          
2017-06-20 21:36:11,278:      2 0.103 0.202 0.223 0.182 0.136 0.070                                                                                    
2017-06-20 21:36:11,278:      3 0.051 0.122 0.182 0.197 0.147 0.117 0.074                                                                              
2017-06-20 21:36:11,279:      4       0.067 0.136 0.147 0.182 0.148 0.098 0.066                                                                        
2017-06-20 21:36:11,279:      5             0.070 0.117 0.148 0.133 0.124 0.090 0.067                                                                  
2017-06-20 21:36:11,279:      6                   0.074 0.098 0.124 0.139 0.104 0.080                                                                  
2017-06-20 21:36:11,279:      7                         0.066 0.090 0.104 0.117 0.085                                                                  
2017-06-20 21:36:11,279:      8                               0.067 0.080 0.085 0.089 0.059 0.054 0.060 0.054 0.062 0.051 0.054 0.052 0.051 0.058 0.057
2017-06-20 21:36:11,279:      9                                                 0.059 0.071 0.072 0.080 0.085 0.074 0.073 0.081 0.078 0.072 0.079 0.081
2017-06-20 21:36:11,279:     10                                                 0.054 0.072 0.077 0.076 0.076 0.080 0.073 0.074 0.071 0.076 0.074 0.083
2017-06-20 21:36:11,280:     11                                                 0.060 0.080 0.076 0.070 0.087 0.070 0.078 0.069 0.077 0.083 0.071 0.071
2017-06-20 21:36:11,280:     12                                                 0.054 0.085 0.076 0.087 0.068 0.071 0.079 0.075 0.073 0.076 0.077 0.074
2017-06-20 21:36:11,280:     13                                                 0.062 0.074 0.080 0.070 0.071 0.081 0.076 0.078 0.078 0.080 0.072 0.069
2017-06-20 21:36:11,280:     14                                                 0.051 0.073 0.073 0.078 0.079 0.076 0.083 0.075 0.073 0.083 0.078 0.079
2017-06-20 21:36:11,280:     15                                                 0.054 0.081 0.074 0.069 0.075 0.078 0.075 0.082 0.080 0.076 0.071 0.082
2017-06-20 21:36:11,280:     16                                                 0.052 0.078 0.071 0.077 0.073 0.078 0.073 0.080 0.082 0.072 0.084 0.074
2017-06-20 21:36:11,280:     17                                                 0.051 0.072 0.076 0.083 0.076 0.080 0.083 0.076 0.072 0.066 0.075 0.080
2017-06-20 21:36:11,281:     18                                                 0.058 0.079 0.074 0.071 0.077 0.072 0.078 0.071 0.084 0.075 0.078 0.076
2017-06-20 21:36:11,281:     19                                                 0.057 0.081 0.083 0.071 0.074 0.069 0.079 0.082 0.074 0.080 0.076 0.071
2017-06-20 21:36:11,281: Perron eigenvalue is   0.85404; state equilibration timescale is ~ 6.9 iterations
2017-06-20 21:36:11,281: Computing covariance matrix...
2017-06-20 21:36:11,353: Deltaf_ij:
2017-06-20 21:36:11,353:    0.000 -40.096 -69.388 -85.235 -85.745 -69.520 -35.481  17.262  89.511 295.727 295.727 295.727 295.727 295.727 295.727 295.727 295.727 295.727 295.727 295.727
2017-06-20 21:36:11,353:   40.096   0.000 -29.292 -45.138 -45.648 -29.424   4.615  57.359 129.608 335.823 335.823 335.823 335.823 335.823 335.823 335.823 335.823 335.823 335.823 335.823
2017-06-20 21:36:11,353:   69.388  29.292   0.000 -15.846 -16.357  -0.132  33.907  86.650 158.900 365.115 365.115 365.115 365.115 365.115 365.115 365.115 365.115 365.115 365.115 365.115
2017-06-20 21:36:11,353:   85.235  45.138  15.846   0.000  -0.510  15.714  49.753 102.497 174.746 380.961 380.961 380.961 380.961 380.961 380.961 380.961 380.961 380.961 380.961 380.961
2017-06-20 21:36:11,354:   85.745  45.648  16.357   0.510   0.000  16.225  50.264 103.007 175.256 381.472 381.472 381.472 381.472 381.472 381.472 381.472 381.472 381.472 381.472 381.472
2017-06-20 21:36:11,354:   69.520  29.424   0.132 -15.714 -16.225   0.000  34.039  86.782 159.031 365.247 365.247 365.247 365.247 365.247 365.247 365.247 365.247 365.247 365.247 365.247
2017-06-20 21:36:11,354:   35.481  -4.615 -33.907 -49.753 -50.264 -34.039   0.000  52.743 124.993 331.208 331.208 331.208 331.208 331.208 331.208 331.208 331.208 331.208 331.208 331.208
2017-06-20 21:36:11,354:  -17.262 -57.359 -86.650-102.497-103.007 -86.782 -52.743   0.000  72.249 278.465 278.465 278.465 278.465 278.465 278.465 278.465 278.465 278.465 278.465 278.465
2017-06-20 21:36:11,354:  -89.511-129.608-158.900-174.746-175.256-159.031-124.993 -72.249   0.000 206.216 206.216 206.216 206.216 206.216 206.216 206.216 206.216 206.216 206.216 206.216
2017-06-20 21:36:11,354: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,355: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,356: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,356: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,356: -295.727-335.823-365.115-380.961-381.472-365.247-331.208-278.465-206.216   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,356: dDeltaf_ij:
2017-06-20 21:36:11,356:    0.000   0.013   0.020   0.024   0.027   0.029   0.031   0.033   0.035   0.037   0.037   0.037   0.037   0.037   0.037   0.037   0.037   0.037   0.037   0.037
2017-06-20 21:36:11,356:    0.013   0.000   0.008   0.014   0.018   0.021   0.024   0.026   0.028   0.031   0.031   0.031   0.031   0.031   0.031   0.031   0.031   0.031   0.031   0.031
2017-06-20 21:36:11,356:    0.020   0.008   0.000   0.006   0.011   0.016   0.019   0.022   0.024   0.027   0.027   0.027   0.027   0.027   0.027   0.027   0.027   0.027   0.027   0.027
2017-06-20 21:36:11,357:    0.024   0.014   0.006   0.000   0.006   0.010   0.014   0.018   0.020   0.023   0.023   0.023   0.023   0.023   0.023   0.023   0.023   0.023   0.023   0.023
2017-06-20 21:36:11,357:    0.027   0.018   0.011   0.006   0.000   0.005   0.010   0.013   0.016   0.020   0.020   0.020   0.020   0.020   0.020   0.020   0.020   0.020   0.020   0.020
2017-06-20 21:36:11,357:    0.029   0.021   0.016   0.010   0.005   0.000   0.005   0.009   0.012   0.017   0.017   0.017   0.017   0.017   0.017   0.017   0.017   0.017   0.017   0.017
2017-06-20 21:36:11,357:    0.031   0.024   0.019   0.014   0.010   0.005   0.000   0.004   0.008   0.013   0.013   0.013   0.013   0.013   0.013   0.013   0.013   0.013   0.013   0.013
2017-06-20 21:36:11,357:    0.033   0.026   0.022   0.018   0.013   0.009   0.004   0.000   0.004   0.010   0.010   0.010   0.010   0.010   0.010   0.010   0.010   0.010   0.010   0.010
2017-06-20 21:36:11,357:    0.035   0.028   0.024   0.020   0.016   0.012   0.008   0.004   0.000   0.006   0.006   0.006   0.006   0.006   0.006   0.006   0.006   0.006   0.006   0.006
2017-06-20 21:36:11,357:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,358:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,359:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,359:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,359:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
2017-06-20 21:36:11,359:    0.037   0.031   0.027   0.023   0.020   0.017   0.013   0.010   0.006   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000   0.000
```
It looks like that error is being thrown during the computation of enthalpy expectations. I wonder if some of the energies were `nan`?

**Comment by jchodera** [comment]
Looking at `/cbio/jclab/home/misik/hsa_yank/implicit/hsa_dansylamide_i02/output/experiments/solvent.nc`, it doesn't look like the `energies` variable has any NaNs...

**Comment by MehtapIsik** [comment]
@jchodera This is stdout prints. I saved them in a file called `20170620_analysis_output.log`.

**Comment by Lnaden** [comment]
As I suspected in the slack, (and can confirm from some other searches), this is actually a NumPy error which was thrown starting in version 1.11. 

It also looks like I have a small bug in the code in MBAR there anyways I should fix.

> ...is not printing out the number of uncorrelated samples...

It looks like I am missed that logger entry when I re-factored the analyze code, there was not a very clean way to show it with the public API right now without re-calculating it. not too hard of a fix, but its missing right now. You could look at the jupyter notebook, it would have it.

**Comment by MehtapIsik** [comment]
@Lnaden Can you point the notebook you mentioned to me?

**Comment by Lnaden** [comment]
`yank analyze report (-s STORE | --store=STORE) (-o REPORT | --output=REPORT)`

`-s` is the target directory, `-o` is the output, then you just run the cells like you would a normal Jupyter notebook. The same error will probably be thrown, but it should at least get through the decelerated data block

**Comment by MehtapIsik** [comment]
Thanks! I wasn't aware of this option.

**Comment by andrrizzi** [comment]
This should have been fixed with pymbar 3.0.2
